include/pybind11/detail/class.h

```diff
@@ -28,7 +28,20 @@ inline std::string get_fully_qualified_tp_name(PyTypeObject *type) {
 #if !defined(PYPY_VERSION)
     return type->tp_name;
 #else
-    return handle((PyObject *) type).attr("__module__").cast<std::string>() + "." + type->tp_name;
+    auto module_name = handle((PyObject *) type).attr("__module__").cast<std::string>();
+    bool is_builtin_module = module_name ==
+#  if PY_MAJOR_VERSION >= 3
```

Comment:
How about defining `PYBIND11_BUILTIN` in `detail/common.h` around like ~180?

Reply:
Oh, yes, good plan. I'll do that!

Reply:
That makes a huge difference in the way things look! :-)